polaris/dataset/_base.py

```diff
@@ -175,31 +174,48 @@ def zarr_root(self) -> zarr.Group | None:
         saved_on_hub = self.zarr_root_path.startswith(StorageSession.polaris_protocol)
 
         if self._warn_about_remote_zarr:
-            saved_remote = saved_on_hub or not Path(self.zarr_root_path).exists()
-
-            if saved_remote:
+            if saved_on_hub:
+                # TODO (cwognum): The user now has no easy way of knowing whether the dataset is "small enough".
                 logger.warning(
                     f"You're loading data from a remote location. "
-                    f"To speed up this process, consider caching the dataset first "
-                    f"using {self.__class__.__name__}.cache()"
+                    f"If the dataset is small enough, consider caching the dataset first "
+                    f"using {self.__class__.__name__}.cache() for more performant data access."
                 )
                 self._warn_about_remote_zarr = False
 
         try:
             if saved_on_hub:
-                with PolarisHubClient() as client:
-                    with StorageSession(client, "read", self.urn) as storage:
-                        self._zarr_root = zarr.open_consolidated(storage.extension_store)
+                self._zarr_root = self.load_zarr_root_from_hub()
```

Comment:
**Comment**
This is nice, much cleaner. Getting closer to a repository pattern.